src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,719 @@
+    public long getTimeUntilNextRun() {
+        return 0;
+    }
+
+    private enum ProbeResult
+    {
+        kBitrateUpdated(0),
+        kNoUpdate(1);
+        int value;
+        ProbeResult(int x)
+        {
+            this.value = x;
+        };
+    }
+
+    private final Object critSect = new Object();
+    private Collection<Integer> ssrcs = new ArrayList<Integer>();
+    private final static int kTimestampGroupLengthMs = 5;
+    private final static int kAbsSendTimeFraction = 18;
+    private final static int kAbsSendTimeInterArrivalUpshift = 8;
+    private final static int kInterArrivalShift
+            =  kAbsSendTimeFraction + kAbsSendTimeInterArrivalUpshift;
+    private final static int kInitialProbingIntervalMs = 2000;
+    private final static int kMinClusterSize = 4;
+    private final static int kMaxProbePackets = 15;
+    private final static int kExpectedNumberOfProbes = 3;
+
+    private static final double kTimestampToMs = 1000.0 /
+            (1 << kInterArrivalShift) ;
+
+    private <K,V> List<K> Keys(TreeMap<K,V> _map)
+    {
+        ArrayList<K> keys = new ArrayList<K>();
+        for(Map.Entry<K,V> entry : _map.entrySet())
+        {
+            keys.add(entry.getKey());
+        }
+        return keys;
+    }
+
+    private long ConvertMsTo24Bits(long timeMs)
+    {
+        long time24Bits = (long)(((timeMs << kAbsSendTimeFraction) + 500) /
+                1000) & 0x00FFFFFF;
+        return time24Bits;
+    }
+
+    private  TreeMap<Long,Long> ssrcs_ = new TreeMap<Long, Long>();
+    private  ArrayList<Probe> probes_ = new ArrayList<>();
+    private  long totalProbesReceived;
+    private  long firstPacketTimeMs;
+    private  long lastUpdateMs;
+    public RemoteBitrateObserver observer_;
+    private AimdRateControl remoteRate  = new AimdRateControl();
+    private InterArrival interArrival;
+    private OveruseEstimator estimator;
+    private OveruseDetector detector;
+    private OverUseDetectorOptions options;
+    private RateStatistics incomingBitrate ;
+    boolean incomingBitrateInitialized;
+    ArrayList<Integer> recentPropagationDeltaMs;
+    ArrayList<Long> recentUpdateTimeMs;
+    private AbsSendTimeEngine absoluteSendTimeEngine = new AbsSendTimeEngine();
+
+    boolean IsWithinClusterBounds(long sendDeltaMs, Cluster clusterAggregate)
+    {
+        if(clusterAggregate.count == 0)
+        {
+            return true;
+        }
+        double clusterMean = clusterAggregate.sendMeanMs /
+                (double)clusterAggregate.count;
+        return  Math.abs((double)sendDeltaMs - clusterMean) < 2.5f;
+    }
+
+    void addCluster(List<Cluster> clusters, Cluster cluster)
+    {
+        cluster.sendMeanMs /= (double)cluster.count;
+        cluster.recvMeanMs /= (double)cluster.count;
+        cluster.meanSize /= cluster.count;
+        clusters.add(cluster);
+    }
+
+    public RemoteBitrateEstimatorAbsSendTime(RemoteBitrateObserver observer)
+    {
+        this.observer_ = observer;
```

Comment:
Call super(RTPPacketPredicate.INSTANCE), so that reverseTransform is only passed RTP packets. Move the constructor to the proper place.